### ISSUE TITLE
Current Provider Filter for Cell Tower Map

### ISSUE THREAD

**Issue by Marhasp** [christianrowlands/android-network-survey#56]
Would it be possible to display only the cell towers of the corresponding technology on the map?
So if I use LTE, then only show the towers of LTE. If I am connected to 2G, then the cell towers of 2G. 
I see all towers in the area at once.

**Comment by christianrowlands** [comment]
There is a way to select the technology on the map currently, it is via the blue button in the top right corner. However, I have not wired up the current serving cell technology with auto-selecting that option when you open the map view. Right now it defaults to LTE every time, but it would be nice to have it default to the current serving cell technology.

**Comment by Marhasp** [comment]
I think I have confused something.
Would it be possible to display only the base stations from the provider you are connected to?
If I have LTE, then only the LTE base stations of the current provider are displayed.

**Comment by christianrowlands** [comment]
Ahh, yeah. I could add a provider filter based on the MCC/MNC.

**Comment by Marhasp** [comment]
Exactly. For example, I use LTE from provider xyz and only get the corresponding cell ids from the provider xyz and LTE.

**Comment by christianrowlands** [comment]
@Marhasp I was able to complete the PLMN filter option for the tower map. The filter is automatically set using the current serving cell PLMN (MCC/MNC), but you also have an option to change it. This feature will come out in the next release. I am closing this issue, but feel free to comment further if you have any problems with the filter or any suggestion for improvement.

One quick note, if you open up the map too fast before the serving cell is discovered, then the filter is blank. I might look into a way to update it after, but I thought it might be confusing if the filter is changed out from under the user after they are already in the map.

Here is a screenshot of the new button at the top and the setting dialog:

![Image](https://github.com/user-attachments/assets/e757d206-5438-4450-b1a6-6c60b502aaf3)

**Comment by Timorttr** [comment]
Thank you. Very good.
I have a dual-sim cell phone. Would it be possible to automatically change the filter for the map depending on the sim card when switching from sim 1 to sim 2?

**Comment by Timorttr** [comment]
Example:
Sim 1 123 2 LTE -> Sim 2 123 1 GSM
